### DISCUSSION TITLE
Documentation Wrong for useAtomCallback?

### DISCUSSION THREAD

**Discussion by stevemolitor** [pmndrs/jotai#2149] [Bug report] [answered]
The [documentation](https://jotai.org/docs/utilities/callback#usage) for `useAtomCallback` says that it returns a promise:

```ts
useAtomCallback(
  callback: (get: Getter, set: Setter, arg: Arg) => Result
): (arg: Arg) => Promise<Result>
```

However the [source code](https://github.com/pmndrs/jotai/blob/main/src/react/utils/useAtomCallback.ts#L8-L11) says it just returns whatever the `Result` of the callback functions is, not necessarily a promise:

```ts
export function useAtomCallback<Result, Args extends unknown[]>(
  callback: (get: Getter, set: Setter, ...arg: Args) => Result,
  options?: Options
): (...args: Args) => Result {
```

Is the documentation wrong here? Maybe this got missed with the v2 changes?

Thanks!

**Comment by dai-shi** [comment, marked as answer]
> Maybe this got missed with the v2 changes?

That's true. Would you like to open a PR to fix it?

**Reply by stevemolitor** [reply]
Will do! At work atm but will do later.

What's the process for testing doc changes? I guess I run the gatsby [dev](https://github.com/pmndrs/jotai/blob/main/website/package.json#L13) script in the `website` package to test doc change locally first?

**Reply by dai-shi** [reply]
You need to run `yarn run prettier` to format docs. Other than that, we basically review changes on a PR.

**Reply by stevemolitor** [reply]
Cool will do.

**Comment by stevemolitor** [comment]
I created a PR - #2150.

I forked the CodeSanbox and updated it as well, but wasn't sure how to update the CodeSandbox link in the docs. So I marked it as Draft for now.
